File: app/src/main/java/com/microlabs/trallet/SettingsActivity.java
package com.microlabs.trallet;

import android.content.DialogInterface;
import android.os.Bundle;
import android.support.v7.app.AlertDialog;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.MenuItem;
import android.view.View;
import android.widget.Toast;

import butterknife.ButterKnife;
import butterknife.OnClick;
import io.realm.Realm;

public class SettingsActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_settings);
        ButterKnife.bind(this);
        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if (item.getItemId() == android.R.id.home) {
            finish();
        }
        return super.onOptionsItemSelected(item);
    }

    @OnClick({R.id.btnDeleteAll, R.id.btnAbout})
    public void onClick(View view) {
        switch (view.getId()) {
            case R.id.btnDeleteAll:
                AlertDialog.Builder builder = new AlertDialog.Builder(this);
                builder.setMessage("Are you sure want to delete all data?");
                builder.setPositiveButton("Yes", new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialog, int which) {
                        Realm realm = Realm.getDefaultInstance();
                        realm.beginTransaction();
                        realm.deleteAll();
                        realm.commitTransaction();
                        Toast.makeText(SettingsActivity.this, "All data deleted", Toast.LENGTH_SHORT).show();
                        finish();
                    }
                });
                builder.setTitle("Delete all data?");
                builder.setNegativeButton("No", new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialog, int which) {
                        dialog.dismiss();
                    }
                });
                builder.create().show();
                break;
            case R.id.btnAbout:
                AlertDialog.Builder builder1 = new AlertDialog.Builder(this);
                builder1.setTitle("About Us");
                builder1.setMessage("Made by MicroLabs Developers\nAJ\nNA\nNH\nHP");
                builder1.setPositiveButton("OK", new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialog, int which) {
                        dialog.dismiss();
                    }
                });
                builder1.create().show();
                break;
        }
    }
}

File: app/src/main/java/com/microlabs/trallet/SettingsActivity.kt
package com.microlabs.trallet

import android.os.Bundle
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.app.AppCompatDelegate
import androidx.databinding.DataBindingUtil
import com.microlabs.trallet.databinding.ActivitySettingsBinding
import com.microlabs.trallet.repo.AppDatabase
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.GlobalScope
import kotlinx.coroutines.launch
import org.jetbrains.anko.*

class SettingsActivity : AppCompatActivity() {

    private lateinit var binding: ActivitySettingsBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_settings)
        setSupportActionBar(binding.toolbar)
        supportActionBar!!.setDisplayHomeAsUpEnabled(true)

        binding.content.btnDeleteAll.setOnClickListener {
            alert("Are you sure want to delete all data?", "Delete all data?") {
                yesButton {
                    GlobalScope.launch(Dispatchers.IO) {
                        AppDatabase.getInstance(applicationContext).bookDao().deleteAllBooks()
                    }
//                    Realm.getDefaultInstance().use {
//                        it.deleteAll()
//                    }
                    toast("All data deleted").show()
                    finish()
                }
                noButton {
                    it.dismiss()
                }
            }.show()
        }

        binding.content.btnAbout.setOnClickListener {
            alert("Made by MicroLabs Developers\n" +
                    "AJ\n" +
                    "NA\n" +
                    "NH\n" +
                    "HP", "About Us") {
                okButton {
                    it.dismiss()
                }
            }.show()
        }

        binding.content.rbGroup.check(when (AppCompatDelegate.getDefaultNightMode()) {
            AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM -> R.id.rbSystem
            AppCompatDelegate.MODE_NIGHT_NO -> R.id.rbLight
            AppCompatDelegate.MODE_NIGHT_YES -> R.id.rbDark
            else -> R.id.rbSystem
        })

        binding.content.rbGroup.setOnCheckedChangeListener { group, checkedId ->
            when (checkedId) {
                R.id.rbSystem -> {
                    AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM)
                }
                R.id.rbDark -> {
                    AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_YES)
                }
                R.id.rbLight -> {
                    AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_NO)
                }
                else -> {
                }
            }
        }
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == android.R.id.home) finish()
        return super.onOptionsItemSelected(item)
    }
}
